src/typescript/frontend/src/configs/local-storage-keys.ts

```diff
@@ -19,20 +31,26 @@ export const LOCAL_STORAGE_CACHE_TIME = {
 export type LocalStorageCache<T> = {
   expiry: number;
   data: T | null;
+  version: string | undefined;
 };
 
-/**
- * Note that this data is not validated and any change in data type returned from this function
- * should be validated to ensure that persisted cache data between multiple builds can cause errors
- * with unexpected data types.
- */
 export function readLocalStorageCache<T>(key: keyof typeof LOCAL_STORAGE_KEYS): T | null {
   const str = localStorage.getItem(LOCAL_STORAGE_KEYS[key]);
   if (str === null) {
     return null;
   }
   try {
     const cache = parseJSON<LocalStorageCache<T>>(str);
+    const range = `~${LOCAL_STORAGE_VERSIONS[key].major}`;
+    // Check for no breaking changes.
+    if (!satisfies(cache.version ?? "1.0.0", range)) {
```

Comment:
The only thing here I'd consider is whether or not this could accidentally default to 1.0.0 unintentionally- say, we forget to add the cache down the road. Since it defaults to 1.0.0, I mean.

But it's not enough to block this, so we can consider it later